❌ **Build Blocked:** Step 1/1 (`Address review feedback`) was **not committed** because build errors could not be resolved after 3 fix attempts + full code regeneration.

This step needs manual review or will be addressed in a follow-up.

<details>
<summary>Build Errors (last attempt)</summary>

```
C:\Agents\principalengineer-e61e56baa444432ebe05b5203cbf0a6f\ReportingDashboard\tests\ReportingDashboard.Tests\Unit\Components\ErrorPanelTests.cs(15,25): error CS1061: 'ErrorPanel' does not contain a definition for 'Message' and no accessible extension method 'Message' accepting a first argument of type 'ErrorPanel' could be found (are you missing a using directive or an assembly reference?) [C:\Agents\principalengineer-e61e56baa444432ebe05b5203cbf0a6f\ReportingDashboard\tests\ReportingDashboard.Tests\ReportingDashboard.Tests.csproj]
C:\Agents\principalengineer-e61e56baa444432ebe05b5203cbf0a6f\ReportingDashboard\tests\ReportingDashboard.Tests\Unit\Components\ErrorPanelTests.cs(24,25): error CS1061: 'ErrorPanel' does not contain a definition for 'Message' and no accessible extension method 'Message' accepting a first argument of type 'ErrorPanel' could be found (are you missing a using directive or an assembly reference?) [C:\Agents\principalengineer-e61e56baa444432ebe05b5203cbf0a6f\ReportingDashboard\tests\ReportingDashboard.Tests\ReportingDashboard.Tests.csproj]
C:\Agents\principalengineer-e61e56baa444432ebe05b5203cbf0a6f\ReportingDashboard\tests\ReportingDashboard.Tests\Unit\Components\HeatmapCellTests.cs(15,25): error CS1061: 'HeatmapCell' does not contain a definition for 'Values' and no accessible extension method 'Values' accepting a first argument of type 'HeatmapCell' could be found (are you missing a using directive or an assembly reference?) [C:\Agents\principalengineer-e61e56baa444432ebe05b5203cbf0a6f\ReportingDashboard\tests\ReportingDashboard.Tests\ReportingDashboard.Tests.csproj]
C:\Agents\principalengineer-e61e56baa444432ebe05b5203cbf0a6f\ReportingDashboard\tests\ReportingDashboard.Tests\Unit\Components\HeatmapCellTests.cs(16,25): error CS1061: 'HeatmapCell' does not contain a definition for 'Label' and no accessible extension method 'Label' accepting a first argument of type 'HeatmapCell' could be found (are you missing a using directive or an assembly reference?) [C:\Agents\principalengineer-e61e56baa444432ebe05b5203cbf0a6f\ReportingDashboard\tests\ReportingDashboard.Tests\ReportingDashboard.Tests.csproj]
C:\Agents\principalengineer-e61e56baa444432ebe05b5203cbf0a6f\ReportingDashboard\tests\ReportingDashboard.Tests\Unit\Components\HeatmapCellTests.cs(17,25): error CS1061: 'HeatmapCell' does not contain a definition for 'IsHeader' and no accessible extension method 'IsHeader' accepting a first argument of type 'HeatmapCell' could be found (are you missing a using directive or an assembly reference?) [C:\Agents\principalengineer-e61e56baa444432ebe05b5203cbf0a6f\ReportingDashboard\tests\ReportingDashboard.Tests\ReportingDashboard.Tests.csproj]
C:\Agents\principalengineer-e61e56baa444432ebe05b5203cbf0a6f\ReportingDashboard\tests\ReportingDashboard.Tests\Unit\Components\ErrorPanelTests.cs(33,25): error CS1061: 'ErrorPanel' does not contain a definition for 'Message' and no accessible extension method 'Message' accepting a first argumen...
```
</details>

--- a/src/ReportingDashboard/Components/Pages/Dashboard.razor
+++ b/src/ReportingDashboard/Components/Pages/Dashboard.razor
@@ -1,23 +1,27 @@
 @page "/"
-@using Microsoft.AspNetCore.Components.Web
+@using ReportingDashboard.Models
+@using ReportingDashboard.Services
 @inject DashboardDataService DataService
-@rendermode RenderMode.InteractiveServer
 
-<div class="dashboard-container">
-    @if (DataService.IsError)
-    {
-        <div class="error-banner">
-            <strong>⚠ Data Error:</strong> @DataService.ErrorMessage
-        </div>
-    }
-    else if (DataService.Data is not null)
-    {
+@if (DataService.IsError)
+{
+    <div class="error-panel">
+        <div class="error-icon">⚠</div>
+        <div class="error-title">Dashboard data could not be loaded</div>
+        <div class="error-details">@DataService.ErrorMessage</div>
+        <div class="error-help">Check data.json for errors and restart the application.</div>
+    </div>
+}
+else if (DataService.Data != null)
+{
+    <div class="dashboard-root">
         <Header Data="@DataService.Data" />
-        <Timeline Data="@DataService.Data.Timeline" />
-        <Heatmap HeatmapData="@DataService.Data.Heatmap" Months="@(DataService.Data.Timeline?.Tracks?.FirstOrDefault()?.Milestones?.Select(m => m.Date).ToList() ?? new())" CurrentMonth="@DateTime.Now.ToString("MMM")" />
-    }
-    else
-    {
-        <ErrorPanel Message="Dashboard data is not available" />
-    }
-</div>
+        @if (DataService.Data.Timeline != null)
+        {
+            <Timeline TimelineData="@DataService.Data.Timeline" />
+        }
+        <Heatmap HeatmapData="@DataService.Data.Heatmap"
+                 Months="@DataService.Data.Months"
+                 CurrentMonth="@DataService.Data.CurrentMonth" />
+    </div>
+}

--- a/src/ReportingDashboard/Components/Timeline.razor
+++ b/src/ReportingDashboard/Components/Timeline.razor
@@ -1,153 +1,182 @@
-@* Timeline component - placeholder for T4 implementation *@
-@* SVG timeline rendering will be implemented in the Timeline SVG Component task *@
+@using ReportingDashboard.Models
+@using System.Globalization
+@using Microsoft.AspNetCore.Components
+@using Microsoft.AspNetCore.Components.Rendering
 
-<div class="tl-area">
-    <div class="tl-labels">
-        @if (TimelineData?.Tracks != null)
-        {
+@if (TimelineData?.Tracks != null && TimelineData.Tracks.Count > 0)
+{
+    <div class="tl-area">
+        <!-- Track Labels Sidebar -->
+        <div style="width:230px; flex-shrink:0; display:flex; flex-direction:column; justify-content:space-around; padding:16px 12px 16px 0; border-right:1px solid #E0E0E0;">
             @foreach (var track in TimelineData.Tracks)
             {
-                <div class="tl-label">
-                    <span class="tl-id" style="color:@track.Color;">@track.Name</span>
-                    <span class="tl-name">@track.Label</span>
+                <div style="font-size:12px;">
+                    <span style="font-weight:600; color:@track.Color;">@track.Name</span>
+                    <span style="font-weight:400; color:#444;"> @track.Label</span>
                 </div>
             }
-        }
-    </div>
-    <div class="tl-svg-box">
-        @if (TimelineData?.Tracks != null)
-        {
-            <svg width="1560" height="@SvgHeight" style="overflow:visible;" xmlns="http://www.w3.org/2000/svg">
+        </div>
+
+        <!-- SVG Timeline Container -->
+        <div class="tl-svg-box">
+            <svg width="@SvgWidth" height="@SvgHeight" style="overflow:visible" xmlns="http://www.w3.org/2000/svg">
+
+                <!-- Drop shadow filter for diamond markers -->
                 <defs>
-                    <filter id="sh">
+                    <filter id="sh" x="-20%" y="-20%" width="140%" height="140%">
                         <feDropShadow dx="0" dy="1" stdDeviation="1.5" flood-opacity="0.3" />
                     </filter>
                 </defs>
 
-                @* Month grid lines and labels *@
-                @for (int i = 0; i < MonthLabels.Count; i++)
+                <!-- Month grid lines at 260px intervals -->
+                @for (int i = 0; i < MonthCount; i++)
                 {
-                    var x = i * ColumnWidth;
-                    <line x1="@F(x)" y1="0" x2="@F(x)" y2="@SvgHeight" stroke="#bbb" stroke-opacity="0.4" />
-                    @SvgText(x + 5, 14, MonthLabels[i], "#666", 11, "start", "600")
+                    var xVal = i * GridInterval;
+                    <line x1="@Fmt(xVal)" y1="0" x2="@Fmt(xVal)" y2="@Fmt(SvgHeight)"
+                          stroke="#bbb" stroke-opacity="0.4" />
                 }
 
-                @* Track lines *@
-                @for (int t = 0; t < TimelineData.Tracks.Count; t++)
+                <!-- Month labels -->
+                @for (int i = 0; i < MonthCount; i++)
                 {
-                    var track = TimelineData.Tracks[t];
-                    var y = TrackY(t);
-                    <line x1="0" y1="@F(y)" x2="1560" y2="@F(y)" stroke="@track.Color" stroke-width="3" />
+                    var xVal = i * GridInterval + 5;
+                    var monthLabel = GetMonthLabel(i);
+                    @RenderSvgText(Fmt(xVal), "14", "#666", "11", "600", null, monthLabel)
+                }
 
-                    @* Milestones *@
-                    @if (StartDate.HasValue && EndDate.HasValue)
+                <!-- Horizontal track lines -->
+                @for (int i = 0; i < TimelineData.Tracks.Count; i++)
+                {
+                    var track = TimelineData.Tracks[i];
+                    var yVal = TrackToY(i);
+                    <line x1="0" y1="@Fmt(yVal)" x2="@Fmt(SvgWidth)" y2="@Fmt(yVal)"
+                          stroke="@track.Color" stroke-width="3" />
+                }
+
+                <!-- Milestone markers and labels -->
+                @for (int trackIdx = 0; trackIdx < TimelineData.Tracks.Count; trackIdx++)
+                {
+                    var track = TimelineData.Tracks[trackIdx];
+                    var trackY = TrackToY(trackIdx);
+
+                    @if (track.Milestones != null)
                     {
-                        @foreach (var ms in track.Milestones)
+                        @foreach (var milestone in track.Milestones)
                         {
-                            if (DateTime.TryParse(ms.Date, out var msDt))
+                            var mx = DateToX(milestone.Date);
+                            var labelY = trackY - 16;
+
+                            @if (milestone.Type == "poc")
                             {
-                                var mx = DateToX(msDt);
-                                @if (ms.Type == "poc")
-                                {
-                                    <polygon points="@DiamondPoints(mx, y)" fill="#F4B400" filter="url(#sh)">
-                                        <title>@ms.Label</title>
-                                    </polygon>
-                                    @SvgText(mx, y - 12, ms.Label, "#666", 10, "middle", "400")
-                                }
-                                else if (ms.Type == "production")
-                                {
-                                    <polygon points="@DiamondPoints(mx, y)" fill="#34A853" filter="url(#sh)">
-                                        <title>@ms.Label</title>
-                                    </polygon>
-                                    @SvgText(mx, y + 20, ms.Label, "#666", 10, "middle", "400")
-                                }
-                                else
-                                {
-                                    <circle cx="@F(mx)" cy="@F(y)" r="5" fill="white" stroke="@track.Color" stroke-width="2.5">
-                                        <title>@ms.Label</title>
-                                    </circle>
-                                    @SvgText(mx, y - 10, ms.Label, "#666", 10, "middle", "400")
-                                }
+                                <polygon points="@DiamondPoints(mx, trackY)"
+                                         fill="#F4B400" filter="url(#sh)">
+                                    <title>@milestone.Label</title>
+                                </polygon>
                             }
+                            else if (milestone.Type == "production")
+                            {
+                                <polygon points="@DiamondPoints(mx, trackY)"
+                                         fill="#34A853" filter="url(#sh)">
+                                    <title>@milestone.Label</title>
+                                </polygon>
+                            }
+                            else
+                            {
+                                <circle cx="@Fmt(mx)" cy="@Fmt(trackY)" r="6"
+                                        fill="white" stroke="@track.Color" stroke-width="2.5">
+                                    <title>@milestone.Label</title>
+                                </circle>
+                            }
+
+                            @RenderSvgText(Fmt(mx), Fmt(labelY), "#666", "10", null, "middle", milestone.Label)
                         }
                     }
                 }
 
-                @* NOW line *@
-                @if (NowDate.HasValue && StartDate.HasValue && EndDate.HasValue)
-                {
-                    var nowX = DateToX(NowDate.Value);
-                    <line x1="@F(nowX)" y1="0" x2="@F(nowX)" y2="@SvgHeight" stroke="#EA4335" stroke-width="2" stroke-dasharray="5,3" />
-                    @SvgText(nowX, -2, "NOW", "#EA4335", 10, "middle", "700")
+                <!-- NOW dashed line -->
+                @{
+                    var nowX = DateToX(TimelineData.NowDate);
                 }
+                <line x1="@Fmt(nowX)" y1="0" x2="@Fmt(nowX)" y2="@Fmt(SvgHeight)"
+                      stroke="#EA4335" stroke-width="2" stroke-dasharray="5,3" />
+                @RenderSvgText(Fmt(nowX), "10", "#EA4335", "10", "700", "middle", "NOW")
+
             </svg>
-        }
+        </div>
     </div>
-</div>
+}
 
 @code {
     [Parameter]
     public TimelineData? TimelineData { get; set; }
 
-    private DateTime? StartDate => DateTime.TryParse(TimelineData?.StartDate, out var d) ? d : null;
-    private DateTime? EndDate => DateTime.TryParse(TimelineData?.EndDate, out var d) ? d : null;
-    private DateTime? NowDate => DateTime.TryParse(TimelineData?.NowDate, out var d) ? d : null;
+    private const int SvgWidth = 1560;
+    private const int GridInterval = 260;
+    private const int MonthCount = 6;
 
-    private List<string> MonthLabels => TimelineData?.Tracks.Count > 0
-        ? GetMonthLabels()
-        : new List<string>();
+    private int SvgHeight => TimelineData?.Tracks != null
+        ? Math.Max(185, TimelineData.Tracks.Count * 56)
+        : 185;
 
-    private int SvgHeight => Math.Max(185, (TimelineData?.Tracks.Count ?? 3) * 56);
+    private DateTime StartDate => ParseDate(TimelineData?.StartDate);
+    private DateTime EndDate => ParseDate(TimelineData?.EndDate);
 
-    private double ColumnWidth => MonthLabels.Count > 0 ? 1560.0 / MonthLabels.Count : 260.0;
-
-    private List<string> GetMonthLabels()
+    private double DateToX(string? isoDate)
     {
-        if (StartDate is null || EndDate is null) return new();
-        var labels = new List<string>();
-        var current = new DateTime(StartDate.Value.Year, StartDate.Value.Month, 1);
-        while (current <= EndDate.Value)
-        {
-            labels.Add(current.ToString("MMM"));
-            current = current.AddMonths(1);
-        }
-        return labels;
-    }
-
-    private double TrackY(int trackIndex)
-    {
-        var count = TimelineData?.Tracks.Count ?? 1;
-        var spacing = (SvgHeight - 30.0) / count;
-        return 42 + trackIndex * spacing;
-    }
-
-    private double DateToX(DateTime date)
-    {
-        if (StartDate is null || EndDate is null) return 0;
-        var totalDays = (EndDate.Value - StartDate.Value).TotalDays;
+        var date = ParseDate(isoDate);
+        double totalDays = (EndDate - StartDate).TotalDays;
         if (totalDays <= 0) return 0;
-        return ((date - StartDate.Value).TotalDays / totalDays) * 1560;
+        double elapsed = (date - StartDate).TotalDays;
+        return (elapsed / totalDays) * SvgWidth;
     }
 
-    private string DiamondPoints(double cx, double cy, double size = 8)
+    private double TrackToY(int trackIndex)
     {
-        return $"{F(cx - size)},{F(cy)} {F(cx)},{F(cy - size)} {F(cx + size)},{F(cy)} {F(cx)},{F(cy + size)}";
+        int trackCount = TimelineData?.Tracks?.Count ?? 1;
+        double spacing = (double)SvgHeight / (trackCount + 1);
+        return spacing * (trackIndex + 1);
     }
 
-    // Format double without trailing zeros for clean SVG output
-    private static string F(double v) => v.ToString("0.##", System.Globalization.CultureInfo.InvariantCulture);
+    private string DiamondPoints(double cx, double cy, double size = 11)
+    {
+        return FormattableString.Invariant(
+            $"{cx},{cy - size} {cx + size},{cy} {cx},{cy + size} {cx - size},{cy}");
+    }
 
-    // Render SVG <text> as RenderFragment to avoid Razor <text> directive conflict
-    private RenderFragment SvgText(double x, double y, string content, string fill, int fontSize, string anchor, string weight) => builder =>
+    private string GetMonthLabel(int gridIndex)
+    {
+        var monthDate = StartDate.AddMonths(gridIndex);
+        return monthDate.ToString("MMM", CultureInfo.InvariantCulture);
+    }
+
+    private static string Fmt(double value) =>
+        value.ToString("0.##", CultureInfo.InvariantCulture);
+
+    private static string Fmt(int value) =>
+        value.ToString(CultureInfo.InvariantCulture);
+
+    private static DateTime ParseDate(string? isoDate)
+    {
+        if (string.IsNullOrEmpty(isoDate)) return DateTime.Today;
+        return DateTime.TryParse(isoDate, CultureInfo.InvariantCulture,
+            System.Globalization.DateTimeStyles.None, out var dt)
+            ? dt
+            : DateTime.Today;
+    }
+
+    private RenderFragment RenderSvgText(string x, string y, string fill, string fontSize, string? fontWeight, string? textAnchor, string content) => builder =>
     {
         builder.OpenElement(0, "text");
-        builder.AddAttribute(1, "x", F(x));
-        builder.AddAttribute(2, "y", F(y));
+        builder.AddAttribute(1, "x", x);
+        builder.AddAttribute(2, "y", y);
         builder.AddAttribute(3, "fill", fill);
-        builder.AddAttribute(4, "font-size", fontSize.ToString());
-        builder.AddAttribute(5, "text-anchor", anchor);
-        builder.AddAttribute(6, "font-weight", weight);
-        builder.AddContent(7, content);
+        builder.AddAttribute(4, "font-size", fontSize);
+        if (fontWeight != null)
+            builder.AddAttribute(5, "font-weight", fontWeight);
+        if (textAnchor != null)
+            builder.AddAttribute(6, "text-anchor", textAnchor);
+        builder.AddAttribute(7, "font-family", "'Segoe UI', Arial, sans-serif");
+        builder.AddContent(8, content);
         builder.CloseElement();
     };
 }

--- a/src/ReportingDashboard/Models/DashboardData.cs
+++ b/src/ReportingDashboard/Models/DashboardData.cs
@@ -2,169 +2,29 @@ using System.Text.Json.Serialization;
 
 namespace ReportingDashboard.Models;
 
-public record DashboardData
+public class DashboardData
 {
     [JsonPropertyName("title")]
-    public string Title { get; init; } = string.Empty;
+    public string Title { get; set; } = string.Empty;
 
     [JsonPropertyName("subtitle")]
-    public string Subtitle { get; init; } = string.Empty;
+    public string Subtitle { get; set; } = string.Empty;
 
     [JsonPropertyName("backlogLink")]
-    public string BacklogLink { get; init; } = string.Empty;
+    public string BacklogLink { get; set; } = string.Empty;
 
     [JsonPropertyName("currentMonth")]
-    public string CurrentMonth { get; init; } = string.Empty;
+    public string CurrentMonth { get; set; } = string.Empty;
 
     [JsonPropertyName("months")]
-    public List<string> Months { get; init; } = [];
-
-    [JsonPropertyName("project")]
-    public ProjectInfo? Project { get; init; }
-
-    [JsonPropertyName("milestones")]
-    public List<Milestone> Milestones { get; init; } = [];
-
-    [JsonPropertyName("shipped")]
-    public List<WorkItem> Shipped { get; init; } = [];
-
-    [JsonPropertyName("inProgress")]
-    public List<WorkItem> InProgress { get; init; } = [];
-
-    [JsonPropertyName("carriedOver")]
-    public List<WorkItem> CarriedOver { get; init; } = [];
-
-    [JsonPropertyName("currentMonthSummary")]
-    public MonthSummary? CurrentMonthSummary { get; init; }
+    public List<string> Months { get; set; } = new();
 
     [JsonPropertyName("timeline")]
-    public TimelineData Timeline { get; init; } = new();
+    public TimelineData? Timeline { get; set; }
 
     [JsonPropertyName("heatmap")]
-    public HeatmapData Heatmap { get; init; } = new();
+    public HeatmapData? Heatmap { get; set; }
 
-    public string? ErrorMessage { get; init; }
-}
-
-public record ProjectInfo
-{
-    [JsonPropertyName("name")]
-    public string Name { get; init; } = "Untitled Project";
-
-    [JsonPropertyName("lead")]
-    public string? Lead { get; init; }
-
-    [JsonPropertyName("status")]
-    public string Status { get; init; } = "Unknown";
-
-    [JsonPropertyName("lastUpdated")]
-    public string? LastUpdated { get; init; }
-
-    [JsonPropertyName("summary")]
-    public string? Summary { get; init; }
-}
-
-public record Milestone
-{
-    [JsonPropertyName("title")]
-    public string Title { get; init; } = "";
-
-    [JsonPropertyName("targetDate")]
-    public string? TargetDate { get; init; }
-
-    [JsonPropertyName("status")]
-    public string Status { get; init; } = "Upcoming";
-}
-
-public record WorkItem
-{
-    [JsonPropertyName("title")]
-    public string Title { get; init; } = "";
-
-    [JsonPropertyName("description")]
-    public string? Description { get; init; }
-
-    [JsonPropertyName("category")]
-    public string? Category { get; init; }
-
-    [JsonPropertyName("percentComplete")]
-    public int PercentComplete { get; init; }
-
-    [JsonPropertyName("carryOverReason")]
-    public string? CarryOverReason { get; init; }
-}
-
-public record MonthSummary
-{
-    [JsonPropertyName("month")]
-    public string? Month { get; init; }
-
-    [JsonPropertyName("totalItems")]
-    public int TotalItems { get; init; }
-
-    [JsonPropertyName("completedItems")]
-    public int CompletedItems { get; init; }
-
-    [JsonPropertyName("carriedItems")]
-    public int CarriedItems { get; init; }
-
-    [JsonPropertyName("overallHealth")]
-    public string OverallHealth { get; init; } = "Unknown";
-}
-
-public record TimelineData
-{
-    [JsonPropertyName("startDate")]
-    public string StartDate { get; init; } = string.Empty;
-
-    [JsonPropertyName("endDate")]
-    public string EndDate { get; init; } = string.Empty;
-
-    [JsonPropertyName("nowDate")]
-    public string NowDate { get; init; } = string.Empty;
-
-    [JsonPropertyName("tracks")]
-    public List<TimelineTrack> Tracks { get; init; } = [];
-}
-
-public record TimelineTrack
-{
-    [JsonPropertyName("name")]
-    public string Name { get; init; } = string.Empty;
-
-    [JsonPropertyName("label")]
-    public string Label { get; init; } = string.Empty;
-
-    [JsonPropertyName("color")]
-    public string Color { get; init; } = "#999";
-
-    [JsonPropertyName("milestones")]
-    public List<TimelineMilestone> Milestones { get; init; } = [];
-}
-
-public record TimelineMilestone
-{
-    [JsonPropertyName("date")]
-    public string Date { get; init; } = string.Empty;
-
-    [JsonPropertyName("type")]
-    public string Type { get; init; } = "checkpoint";
-
-    [JsonPropertyName("label")]
-    public string Label { get; init; } = string.Empty;
-}
-
-public record HeatmapData
-{
-    [JsonPropertyName("shipped")]
-    public Dictionary<string, List<string>> Shipped { get; init; } = new();
-
-    [JsonPropertyName("inProgress")]
-    public Dictionary<string, List<string>> InProgressItems { get; init; } = new();
-
-    [JsonPropertyName("carryover")]
-    public Dictionary<string, List<string>> Carryover { get; init; } = new();
-
-    [JsonPropertyName("blockers")]
-    public Dictionary<string, List<string>> Blockers { get; init; } = new();
+    [JsonIgnore]
+    public string? ErrorMessage { get; set; }
 }

--- a/src/ReportingDashboard/Models/HeatmapData.cs
+++ b/src/ReportingDashboard/Models/HeatmapData.cs
@@ -1,0 +1,22 @@
+using System.Text.Json.Serialization;
+
+namespace ReportingDashboard.Models;
+
+public class HeatmapData
+{
+    [JsonPropertyName("shipped")]
+    public Dictionary<string, List<string>> Shipped { get; set; } = new();
+
+    [JsonPropertyName("inProgress")]
+    public Dictionary<string, List<string>> InProgress { get; set; } = new();
+
+    [JsonPropertyName("carryover")]
+    public Dictionary<string, List<string>> Carryover { get; set; } = new();
+
+    [JsonPropertyName("blockers")]
+    public Dictionary<string, List<string>> Blockers { get; set; } = new();
+
+    /// <summary>Alias for InProgress, used by existing Heatmap.razor component.</summary>
+    [JsonIgnore]
+    public Dictionary<string, List<string>> InProgressItems => InProgress;
+}

--- a/src/ReportingDashboard/Models/MonthSummary.cs
+++ b/src/ReportingDashboard/Models/MonthSummary.cs
@@ -1,0 +1,10 @@
+namespace ReportingDashboard.Models;
+
+public record MonthSummary
+{
+    public string? Month { get; init; }
+    public int TotalItems { get; init; }
+    public int CompletedItems { get; init; }
+    public int CarriedItems { get; init; }
+    public string OverallHealth { get; init; } = "Unknown";
+}

--- a/src/ReportingDashboard/Models/TimelineData.cs
+++ b/src/ReportingDashboard/Models/TimelineData.cs
@@ -1,0 +1,45 @@
+using System.Text.Json.Serialization;
+
+namespace ReportingDashboard.Models;
+
+public class TimelineData
+{
+    [JsonPropertyName("startDate")]
+    public string StartDate { get; set; } = string.Empty;
+
+    [JsonPropertyName("endDate")]
+    public string EndDate { get; set; } = string.Empty;
+
+    [JsonPropertyName("nowDate")]
+    public string NowDate { get; set; } = string.Empty;
+
+    [JsonPropertyName("tracks")]
+    public List<TimelineTrack> Tracks { get; set; } = new();
+}
+
+public class TimelineTrack
+{
+    [JsonPropertyName("name")]
+    public string Name { get; set; } = string.Empty;
+
+    [JsonPropertyName("label")]
+    public string Label { get; set; } = string.Empty;
+
+    [JsonPropertyName("color")]
+    public string Color { get; set; } = "#999";
+
+    [JsonPropertyName("milestones")]
+    public List<Milestone> Milestones { get; set; } = new();
+}
+
+public class Milestone
+{
+    [JsonPropertyName("date")]
+    public string Date { get; set; } = string.Empty;
+
+    [JsonPropertyName("type")]
+    public string Type { get; set; } = "checkpoint";
+
+    [JsonPropertyName("label")]
+    public string Label { get; set; } = string.Empty;
+}

--- a/src/ReportingDashboard/wwwroot/css/dashboard.css
+++ b/src/ReportingDashboard/wwwroot/css/dashboard.css
@@ -1,42 +1,25 @@
-:root {
-    --color-completed: #2e7d32;
-    --color-in-progress: #1565c0;
-    --color-at-risk: #ef6c00;
-    --color-carried-over: #ef6c00;
-    --color-upcoming: #757575;
-    --color-behind: #c62828;
-    --color-on-track: #2e7d32;
-    --color-bg: #ffffff;
-    --color-bg-alt: #f5f5f5;
-    --color-text: #212121;
-    --color-text-secondary: #616161;
-    --color-border: #e0e0e0;
-    --font-stack: 'Segoe UI', Arial, sans-serif;
-    --max-width: 1200px;
-}
-
 * { box-sizing: border-box; margin: 0; padding: 0; }
 
 body {
-    font-family: var(--font-stack);
+    font-family: 'Segoe UI', Arial, sans-serif;
     font-size: 14px;
     line-height: 1.5;
-    color: var(--color-text);
-    background: var(--color-bg);
+    color: #111;
+    background: #FFFFFF;
     width: 1920px;
     height: 1080px;
     overflow: hidden;
 }
 
-.dashboard-container {
-    display: flex;
-    flex-direction: column;
+.dashboard-root {
     width: 1920px;
     height: 1080px;
-    background: var(--color-bg);
+    display: flex;
+    flex-direction: column;
+    overflow: hidden;
 }
 
-/* ── Header ─────────────────────────────────────────────── */
+/* ---- Header ---- */
 .hdr {
     display: flex;
     justify-content: space-between;
@@ -50,184 +33,172 @@ body {
     font-weight: 700;
     color: #111;
     margin: 0;
-    line-height: 1.2;
 }
-.hdr .sub {
+.sub {
     font-size: 12px;
     color: #888;
     margin-top: 2px;
 }
-.hdr-legend {
+
+/* ---- SVG Timeline Section ---- */
+.tl-area {
+    height: 196px;
+    background: #FAFAFA;
+    border-bottom: 2px solid #E8E8E8;
+    padding: 6px 44px 0;
     display: flex;
-    gap: 22px;
-    align-items: center;
-    font-size: 12px;
-    color: #444;
-}
-.legend-item {
-    display: flex;
-    align-items: center;
-    gap: 6px;
-    white-space: nowrap;
-}
-.legend-diamond {
-    display: inline-block;
-    width: 12px;
-    height: 12px;
-    transform: rotate(45deg);
-    flex-shrink: 0;
-}
-.legend-dot {
-    display: inline-block;
-    width: 8px;
-    height: 8px;
-    border-radius: 50%;
-    background: #999;
-    flex-shrink: 0;
-}
-.legend-now {
-    display: inline-block;
-    width: 2px;
-    height: 14px;
-    background: #EA4335;
+    align-items: stretch;
     flex-shrink: 0;
 }
 
-/* ── Error Banner ───────────────────────────────────────── */
-.error-banner {
-    background: #fde8e8;
-    border: 1px solid #f5c6c6;
-    border-left: 4px solid var(--color-behind);
-    padding: 16px 20px;
-    border-radius: 4px;
-    margin-bottom: 24px;
-    font-size: 14px;
-}
-
-/* ── Sections (legacy) ──────────────────────────────────── */
-.section {
-    margin-bottom: 24px;
-    padding: 20px;
-    background: var(--color-bg);
-    border: 1px solid var(--color-border);
-    border-radius: 6px;
-}
-
-h2 { font-size: 18px; font-weight: 600; margin-bottom: 16px; color: var(--color-text); }
-
-/* Project Header (legacy) */
-.project-header { border-left: 4px solid var(--color-in-progress); }
-.header-row { display: flex; justify-content: space-between; align-items: flex-start; margin-bottom: 12px; }
-.header-meta { display: flex; gap: 12px; align-items: center; }
-.project-lead { color: var(--color-text-secondary); font-size: 14px; }
-.project-summary { color: var(--color-text-secondary); font-size: 14px; line-height: 1.6; }
-.last-updated { color: var(--color-text-secondary); font-size: 13px; }
-
-.status-badge {
-    display: inline-block;
-    padding: 4px 12px;
-    border-radius: 12px;
-    font-size: 13px;
-    font-weight: 600;
-    color: #fff;
-}
-.status-on-track { background: var(--color-on-track); }
-.status-at-risk { background: var(--color-at-risk); }
-.status-behind { background: var(--color-behind); }
-.status-unknown { background: var(--color-upcoming); }
-
-/* Monthly Summary Metrics */
-.metrics-grid { display: grid; grid-template-columns: repeat(4, 1fr); gap: 16px; }
-.metric-card {
-    text-align: center;
-    padding: 16px;
-    background: var(--color-bg-alt);
-    border-radius: 6px;
-    border: 1px solid var(--color-border);
-}
-.metric-value { display: block; font-size: 28px; font-weight: 700; margin-bottom: 4px; }
-.metric-label { display: block; font-size: 13px; color: var(--color-text-secondary); }
-.health-on-track { color: var(--color-on-track); }
-.health-at-risk { color: var(--color-at-risk); }
-.health-behind { color: var(--color-behind); }
-
-/* Milestone Timeline */
-.timeline-track {
-    display: grid;
-    grid-template-columns: repeat(auto-fit, minmax(150px, 1fr));
-    gap: 8px;
-    position: relative;
-}
-.timeline-item { text-align: center; padding: 12px 8px; border-radius: 6px; background: var(--color-bg-alt); border: 1px solid var(--color-border); }
-.timeline-marker {
-    width: 12px; height: 12px;
-    border-radius: 50%;
-    margin: 0 auto 8px;
-    background: var(--color-upcoming);
-}
-.milestone-completed .timeline-marker { background: var(--color-completed); }
-.milestone-in-progress .timeline-marker { background: var(--color-in-progress); }
-.milestone-at-risk .timeline-marker { background: var(--color-at-risk); }
-.milestone-upcoming .timeline-marker { background: var(--color-upcoming); }
-.milestone-title { display: block; font-size: 13px; font-weight: 600; margin-bottom: 4px; }
-.milestone-date { display: block; font-size: 12px; color: var(--color-text-secondary); }
-.milestone-status { display: block; font-size: 11px; color: var(--color-text-secondary); text-transform: uppercase; letter-spacing: 0.5px; margin-top: 4px; }
-
-/* Work Items */
-.work-items { display: flex; flex-direction: column; gap: 12px; }
-.work-item {
-    padding: 16px;
-    border: 1px solid var(--color-border);
-    border-radius: 6px;
-    border-left: 4px solid var(--color-border);
-    background: var(--color-bg);
-}
-.border-completed { border-left-color: var(--color-completed); }
-.border-in-progress { border-left-color: var(--color-in-progress); }
-.border-carried-over { border-left-color: var(--color-carried-over); }
-
-.item-header { display: flex; justify-content: space-between; align-items: center; margin-bottom: 8px; }
-.item-title { font-size: 15px; font-weight: 600; }
-.item-category {
-    font-size: 12px;
-    padding: 2px 8px;
-    background: var(--color-bg-alt);
-    border: 1px solid var(--color-border);
-    border-radius: 10px;
-    color: var(--color-text-secondary);
-}
-.item-description { font-size: 13px; color: var(--color-text-secondary); margin-bottom: 8px; }
-.carry-over-reason { font-size: 13px; color: var(--color-at-risk); margin-bottom: 8px; }
-
-.progress-bar-container {
-    position: relative;
-    height: 20px;
-    background: var(--color-bg-alt);
-    border-radius: 10px;
+.tl-svg-box {
+    flex: 1;
+    padding-left: 12px;
+    padding-top: 6px;
     overflow: hidden;
-    border: 1px solid var(--color-border);
-}
-.progress-bar {
-    height: 100%;
-    border-radius: 10px;
-    transition: width 0.3s ease;
-}
-.progress-completed { background: var(--color-completed); }
-.progress-in-progress { background: var(--color-in-progress); }
-.progress-carried-over { background: var(--color-carried-over); }
-.progress-label {
-    position: absolute;
-    right: 8px;
-    top: 50%;
-    transform: translateY(-50%);
-    font-size: 11px;
-    font-weight: 600;
-    color: var(--color-text);
 }
 
-/* Print styles */
+/* ---- Heatmap Section ---- */
+.hm-wrap {
+    flex: 1;
+    min-height: 0;
+    display: flex;
+    flex-direction: column;
+    padding: 10px 44px 10px;
+}
+
+.hm-title {
+    font-size: 14px;
+    font-weight: 700;
+    color: #888;
+    letter-spacing: 0.5px;
+    text-transform: uppercase;
+    margin-bottom: 8px;
+    flex-shrink: 0;
+}
+
+.hm-grid {
+    display: grid;
+    flex: 1;
+    min-height: 0;
+    border: 1px solid #E0E0E0;
+}
+
+/* Corner cell */
+.hm-corner {
+    background: #F5F5F5;
+    font-size: 11px;
+    font-weight: 700;
+    color: #999;
+    text-transform: uppercase;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    border-right: 1px solid #E0E0E0;
+    border-bottom: 2px solid #CCC;
+}
+
+/* Column headers */
+.hm-col-hdr {
+    background: #F5F5F5;
+    font-size: 16px;
+    font-weight: 700;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    border-right: 1px solid #E0E0E0;
+    border-bottom: 2px solid #CCC;
+}
+
+/* Current month column header */
+.apr-hdr {
+    background: #FFF0D0 !important;
+    color: #C07700 !important;
+}
+
+/* Row headers */
+.hm-row-hdr {
+    font-size: 11px;
+    font-weight: 700;
+    text-transform: uppercase;
+    letter-spacing: 0.7px;
+    padding: 0 12px;
+    display: flex;
+    align-items: center;
+    border-right: 2px solid #CCC;
+    border-bottom: 1px solid #E0E0E0;
+}
+
+.ship-hdr { color: #1B7A28; background: #E8F5E9; }
+.prog-hdr { color: #1565C0; background: #E3F2FD; }
+.carry-hdr { color: #B45309; background: #FFF8E1; }
+.block-hdr { color: #991B1B; background: #FEF2F2; }
+
+/* Data cells */
+.hm-cell {
+    padding: 8px 12px;
+    border-right: 1px solid #E0E0E0;
+    border-bottom: 1px solid #E0E0E0;
+    overflow: hidden;
+}
+
+/* Shipped cells */
+.ship-cell { background: #F0FBF0; }
+.ship-cell.apr { background: #D8F2DA; }
+
+/* In Progress cells */
+.prog-cell { background: #EEF4FE; }
+.prog-cell.apr { background: #DAE8FB; }
+
+/* Carryover cells */
+.carry-cell { background: #FFFDE7; }
+.carry-cell.apr { background: #FFF0B0; }
+
+/* Blockers cells */
+.block-cell { background: #FFF5F5; }
+.block-cell.apr { background: #FFE4E4; }
+
+/* Work items in cells */
+.it {
+    font-size: 12px;
+    color: #333;
+    padding: 2px 0 2px 12px;
+    line-height: 1.35;
+    position: relative;
+}
+.it::before {
+    content: '';
+    position: absolute;
+    left: 0;
+    top: 7px;
+    width: 6px;
+    height: 6px;
+    border-radius: 50%;
+}
+
+/* Dot colors per row */
+.ship-cell .it::before { background: #34A853; }
+.prog-cell .it::before { background: #0078D4; }
+.carry-cell .it::before { background: #F4B400; }
+.block-cell .it::before { background: #EA4335; }
+
+/* ---- Error Panel ---- */
+.error-panel {
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    justify-content: center;
+    width: 1920px;
+    height: 1080px;
+    text-align: center;
+    gap: 12px;
+    font-family: 'Segoe UI', Arial, sans-serif;
+}
+.error-icon { font-size: 48px; color: #EA4335; }
+.error-title { font-size: 20px; font-weight: 700; color: #333; }
+.error-details { font-size: 14px; color: #666; font-family: monospace; }
+.error-help { font-size: 12px; color: #888; }
+
 @media print {
     body { background: white; }
-    .dashboard-container { max-width: 100%; padding: 0; box-shadow: none; }
-    .section { break-inside: avoid; border: 1px solid #ccc; }
 }

--- a/tests/ReportingDashboard.Tests/ReportingDashboard.Tests.csproj
+++ b/tests/ReportingDashboard.Tests/ReportingDashboard.Tests.csproj
@@ -4,5 +4,14 @@
     <Nullable>enable</Nullable>
     <ImplicitUsings>enable</ImplicitUsings>
     <IsPackable>false</IsPackable>
+    <IsTestProject>true</IsTestProject>
   </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.8.0" />
+    <PackageReference Include="xunit" Version="2.6.6" />
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.5.6">
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+      <PrivateAssets>all</PrivateAssets>
+    </PackageReference>
+  </ItemGroup>
 </Project>

--- a/tests/ReportingDashboard.UITests/ReportingDashboard.UITests.csproj
+++ b/tests/ReportingDashboard.UITests/ReportingDashboard.UITests.csproj
@@ -4,5 +4,14 @@
     <Nullable>enable</Nullable>
     <ImplicitUsings>enable</ImplicitUsings>
     <IsPackable>false</IsPackable>
+    <IsTestProject>true</IsTestProject>
   </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.8.0" />
+    <PackageReference Include="xunit" Version="2.6.6" />
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.5.6">
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+      <PrivateAssets>all</PrivateAssets>
+    </PackageReference>
+  </ItemGroup>
 </Project>

--- a/tests/UITests/Project.UITests.csproj
+++ b/tests/UITests/Project.UITests.csproj
@@ -4,5 +4,14 @@
     <Nullable>enable</Nullable>
     <ImplicitUsings>enable</ImplicitUsings>
     <IsPackable>false</IsPackable>
+    <IsTestProject>true</IsTestProject>
   </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.8.0" />
+    <PackageReference Include="xunit" Version="2.6.6" />
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.5.6">
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+      <PrivateAssets>all</PrivateAssets>
+    </PackageReference>
+  </ItemGroup>
 </Project>

